apps/api-server/src/util/react-check.js

```diff
@@ -0,0 +1,42 @@
+const reactJs =
+  process.env.REACT_CDN ||
+  'https://unpkg.com/react@18.2.0/umd/react.production.min.js';
+const reactDomJs =
+  process.env.REACT_DOM_CDN ||
+  'https://unpkg.com/react-dom@18.2.0/umd/react-dom.production.min.js';
+
+module.exports = `
+  function triggerEvent() {
+    document.dispatchEvent(reactLoadedEvent);
+  }
+  
+  const reactLoadedEvent = new CustomEvent('OpenStadReactLoaded');
+  
+  if (typeof React === 'undefined' && !window.OpenStadReactLoaded) {
+    const script = document.createElement('script');
+    script.src = '${reactJs}';
+    document.body.appendChild(script);
+    window.OpenStadReactLoaded = true;
+  } else if (
+    typeof React !== 'undefined' &&
+    React.version.substr(0, 2) !== '18'
+  ) {
+    throw new Error('React version 18 is required');
+  }
+  
+  if (typeof ReactDOM === 'undefined' && !window.OSReactDomLoaded) {
+    const script = document.createElement('script');
+    script.src = '${reactDomJs}';
```

Comment:
Zie hierboven dezelfde vraag

Reply:
Dit gehele stuk (vanaf `module.exports`) staat in backticks. Omdat we dit niet als code maar als tekst willen teruggeven, zodat het client-side uitgevoerd kan worden.